mantle/platform/qemu.go

```diff
@@ -306,6 +306,41 @@ func (inst *QemuInstance) SwitchBootOrder() (err error) {
 	return nil
 }
 
+// RemoveBlockDevice deletes the primary device from a qemu instance
+// and sets the seconday device as primary.
+func (inst *QemuInstance) RemoveBlockDevice() error {
+	var primaryDevice string
+	var secondaryDevicePath []string
+	monitor, err := newQMPMonitor(inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+	monitor.Connect()
+	defer monitor.Disconnect()
+	blkdevs, err := listQMPBlkDevices(monitor, inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not list block devices through qmp")
+	}
+	for _, dev := range blkdevs.Return {
+		if !dev.Removable {
+			if dev.Inserted.BackingfileDepth == 1 {
+				primaryDevice = dev.DevicePath
+			} else {
+				secondaryDevicePath = append(secondaryDevicePath, dev.DevicePath)
+			}
+		}
+	}
+	err = setBootIndexForDevice(monitor, primaryDevice, -1)
+	if err != nil {
+		return errors.Wrapf(err, "Could not set bootindex for %v", primaryDevice)
+	}
+	err = setBootIndexForDevice(monitor, secondaryDevicePath[0], 1)
```

Comment:
This function should be checking that there are secondary devices to switch the bootindex to. As is this would panic if called in situations where there is only the primary boot device.

Reply:
How about putting an `if` condition to check if the `secondaryDevicePath` exists then set the bootindex, else just return the normal result?

Reply:
If `secondaryDevicePath` doesn't exist this function should return an error (explaining no secondary device was found) before performing any actual operations (e.x. before it sets the primary device boot index to `-1`)